### PR TITLE
Update to latest datastore with contexts

### PR DESCRIPTION
**Pull request by gammazero**
Also updates other dependencies to latest, including libp2p.

### CODE REVIEW

File: internal/ingest/ingest.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 func (ing *Ingester) watchSyncFinished(onSyncFin <-chan legs.SyncFinished) {
 	for syncFin := range onSyncFin {
 		// Persist the latest sync
-		err := ing.ds.Put(datastore.NewKey(syncPrefix+syncFin.PeerID.String()), syncFin.Cid.Bytes())
+		err := ing.ds.Put(context.Background(), datastore.NewKey(syncPrefix+syncFin.PeerID.String()), syncFin.Cid.Bytes())
```
Comment (willscott) [inline internal/ingest/ingest.go:164]: do we want a timeout on these operations?

File: internal/registry/registry.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func NewRegistry(cfg config.Discovery, dstore datastore.Datastore, disco discove
 		dstore: dstore,
 	}
 
-	count, err := r.loadPersistedProviders()
+	count, err := r.loadPersistedProviders(context.Background())
```
Comment (willscott) [inline internal/registry/registry.go:105]: should this one bubble up?

File: internal/registry/registry.go [hunk 3]
```diff
@@ -402,37 +402,36 @@ func (r *Registry) syncStartDiscover(peerID peer.ID, discoAddr string, errCh cha
...
+			if discoErr != nil {
+				errCh <- discoErr
+				return
+			}
+
+			info := &ProviderInfo{
+				AddrInfo:      discoData.AddrInfo,
+				DiscoveryAddr: discoAddr,
+			}
+			if err := r.syncRegister(context.Background(), info); err != nil {
```
Comment (willscott) [inline internal/registry/registry.go:419]: might be good also for this to have a timeout?

File: internal/ingest/ingest.go (hunk 1 above)
Comment (gammazero) [reply internal/ingest/ingest.go:164]: Not for this one.  There is nothing waiting for this to complete, so there is no reason to consider a slow write as too long and need to cancel it.

File: internal/registry/registry.go (hunk 3 above)
Comment (gammazero) [reply internal/registry/registry.go]: This should use the configured discovery timeout.  Fixed.

File: internal/registry/registry.go (hunk 2 above)
Comment (gammazero) [reply internal/registry/registry.go]: Probably makes sense, but only to cancel this if a graceful shutdown is requested while the indexer is still starting up.
Done.

**Review by willscott — APPROVED** [review]

**Pull request merged**